### DISCUSSION TITLE
Handling dynamic connection to a database depending on a request

### DISCUSSION THREAD

**Discussion by maioradv** [prisma/prisma#22790] [Q&A] [answered]
### Question

Hi, i'm using nestjs as backend, i'd like to usa prisma as orm for my project.
My idea is upon a request analyze the host to retrieve from a static database my connection credentials for another database that has been created manually before.
Then with help of prisma i want to initialize that database (if void) or migrate (if needed) and use the prisma client to retrieve data from this database.
Seems like i need to connect with dynamic host, db name etc. and also use the same prisma schema with different databases.
Is it possibile? I can do something similiar? Let me know...

**Comment by ludralph** [comment, marked as answer]
Hi @maioradv 👋 

Thank you for raising this question.

Prisma does support the use of environment variables for database connections, which can be set [programmatically](https://www.prisma.io/docs/orm/reference/prisma-client-reference#programmatically-override-a-datasource-url) at runtime. This could potentially be used to switch between different databases based on the request host. Additionally, Prisma Migrate can be used to manage database migrations, but there is no way to trigger migrations dynamically based on database state because this is not directly supported by Prisma. There is an open feature request [here](https://github.com/prisma/prisma/issues/13549) that you can add a 👍 to. The best workaround for now would be to write scripts that use the CLI commands.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
